### ISSUE TITLE
0.53.0+ fails when config.output is array

### ISSUE THREAD

**Issue by paulsmirnov** [rollup/rollup#1826]
index.js
```js
export default 'foo';
```

rollup.config.js
```js
module.exports = {
  input: 'index.js',
  output: [{
    format: 'umd',
    name: 'foo',
    file: 'foo.js',
  }, {
    format: 'es',
    file: 'foo.module.js',
  }],
};
```

build.js
```js
const rollup = require('rollup').rollup;
const rollupConfig = require('./rollup.config');

rollup(rollupConfig).then((bundle) => {
  return Promise.all(rollupConfig.output.map(function(output) {
    return bundle.write(output);
  }));
});
```

If you run `build.js` you get: ... **Error: Unknown option found: 0, 1.** Allowed keys: input, legacy, treeshake, acorn, context, moduleContext, plugins, onwarn, watch, cache, preferConst, experimentalDynamicImport, entry, external, extend, amd, banner, footer, intro, format, outro, sourcemap, sourcemapFile, name, globals, interop, legacy, freeze, indent, strict, noConflict, paths, exports, file, pureExternalModules.

The error is raised only when calling rollup programmatically. It works ok if run from CLI:

    node_modules\.bin\rollup.cmd -c

**Comment by lukastaegert** [comment]
To my knowledge, having `output` as an array is only part of the config file API but not of the programmatic API as with this API, you could  easily use `.map()` and `Promise.all()` for the same effect but with more fine-grained control. Did this work for older versions? Otherwise, maybe the documentation is misleading or should make this more clear. In any case, the error message could certainly be improved.

**Comment by paulsmirnov** [comment]
This worked in previous versions. Moreover, I've looked into the rollup code now, particularly `rollup()` method. It calls `mergeOptions()` which reads both **input** and **output** options. I don't understand why should it process (and verify) output options if `rollup()` method is nothing to do with output.

If you noticed, I used `map` and `Promise.all()` together with `bundle.write()`, so it WILL work unless `mergeOptions()` throws. Should I explicitly `delete config.output` then, in advance? Confusing 😕

**Comment by paulsmirnov** [comment]
```js
const outputs = rollupConfig.output;
delete rollupConfig.output;
```

This will help me as a temporary workaround but I don't think it should be done so in production.

**Comment by ankeetmaini** [comment]
I'll double check this. And update this issue.

**Comment by ankeetmaini** [comment]
The problem is with [this line](https://github.com/rollup/rollup/blob/master/src/utils/mergeOptions.ts#L163).

```js
const errors = [...Object.keys(config || {}), ...Object.keys(config.output || {})]
.filter(k => k !== 'output' && validKeys.indexOf(k) === -1);
```
This falsely assumes output options is an object, and thus when it's an array it errors out saying `0, 1` as `unrecognized options` which are actually indexes of the array.

Also it fails only programmatically as we're throwing the error only [here](https://github.com/rollup/rollup/blob/master/src/rollup/index.ts#L246). The error was ignored when run via CLI.

**Comment by lukastaegert** [comment]
Closed via #1827 in Rollup@0.53.2
